src/detail/index.jsx

```diff
@@ -1,9 +1,40 @@
 import { useParams } from "react-router";
+import { faWind } from '@fortawesome/free-solid-svg-icons';
+import { faTint } from '@fortawesome/free-solid-svg-icons';
+import { useReducerAPI } from "../api";
+import MiniCard from "../components/minicard";
+import Status from "../components/status";
+
+import "./index.css";
 
 const Detail =function(){
     let { lat,lng } = useParams();
-    
-    return <>Detail</>
+
+    const currentWeatherState = useReducerAPI(
+        `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lng}&units=metric&appid=d79e473c27e1ce85a145c3a51e23f2c5`
+    );
+
+    return(
+        <div className="container">
+            <img
+                src={`http://openweathermap.org/img/wn/${currentWeatherState.data && currentWeatherState.data.current.weather[0].icon}@4x.png`}
+                alt="status" />
+            <p className="capitalize font-medium tracking-wider">{currentWeatherState.data && currentWeatherState.data.current.weather[0].description}</p>
+            <p className="text-7xl my-8">{currentWeatherState.data && currentWeatherState.data.current.temp}&#176;</p>
```

Comment:
Shall we Spread the currentWeatherState.data at top level and do validation there bro ?
```
const currentWeatherState = useReducerAPI(
        `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lng}&units=metric&appid=d79e473c27e1ce85a145c3a51e23f2c5`
    );

if(currentWeatherState && currentWeatherState.data && currentWeatherState.data.current){
  const {temp,wind_speed,humidity}= currentWeatherState.data.current
  
//   do the logic 
}
```